Ufs weather model for pr

SMoorthi-emc opened the pull request:
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [ ] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Provide a detailed description of what this PR does. What bug does it fix, or what feature does it add? Is a change of answers expected from this PR? Are any library updates included in this PR (modulefiles etc.)?

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs

- CCPP-Physics Pull request Sm may182021 [#662](https://github.com/NCAR/ccpp-physics/pull/662)
- fv3atm full request  may182021 [#303](https://github.com/NOAA-EMC/fv3atm/pull/303)

climbfuji commented on tests/compile.sh:
```diff
@@ -54,6 +54,7 @@ BUILD_JOBS=${BUILD_JOBS:-8}
 
 hostname
 
+module purge
```
No, this must be in machine specific files. If you do `module purge` on gaea, for example, the whole system falls apart.

BrianCurtis-NOAA replied:
This is still an issue @SMoorthi-emc we can't module purge on all machines, it breaks Gaea for example.

junwang-noaa commented:
I think it's this script for module purge.

source $PATHTR/NEMS/src/conf/module-setup.sh.inc

On Wed, Jun 2, 2021 at 3:54 PM Dom Heinzeller ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In tests/compile.sh
> <https://github.com/ufs-community/ufs-weather-model/pull/606#discussion_r644276646>
> :
>
> > @@ -54,6 +54,7 @@ BUILD_JOBS=${BUILD_JOBS:-8}
>
>  hostname
>
> +module purge
>
> No, this must be in machine specific files. If you do module purge on
> gaea, for example, the whole system falls apart.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#pullrequestreview-674624688>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TPYAY5QFJXF34HEMITTQ2EAJANCNFSM45U36N4Q>
> .
>

DeniseWorthen reviewed (changes requested):
Several nml.IN files are present in tests/parm but these have been removed from develop. I think they need to be removed.

SMoorthi-emc commented:
Well, I guess the merge does not work right.
I wonder why these files did not get removed when I merged (may be because
in my version I had updated to remove 'ncld")
This is one of the drawbacks of these fast commits I guess.
I don't know which files need to be removed.
Moorthi

On Mon, Jun 21, 2021 at 9:46 AM Denise Worthen ***@***.***>
wrote:

> *@DeniseWorthen* requested changes on this pull request.
>
> Several nml.IN files are present in tests/parm but these have been removed
> from develop. I think they need to be removed.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#pullrequestreview-688442180>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYRMOKLDJXSGDF577S3TT47CPANCNFSM45U36N4Q>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

DeniseWorthen commented:
This is the list of nml.IN currently in develop, tests/parm :

c96_HAFS_v0_hwrf.nml.IN
control_ca.nml.IN
control_csawmg.nml.IN
control_flake.nml.IN
control_gdas.nml.IN
control.nml.IN
control_noahmp.nml.IN
control_rrtmgp.nml.IN
control_thompson.nml.IN
control_ugwpv1.nml.IN
cpt.nml.IN
csawmg3shoc127.nml.IN
csawmgshoc.nml.IN
esg_HAFS_v0_hwrf.nml.IN
gsd.nml.IN
gsd_rrtmgp.nml.IN
gsd_sar.nml.IN
input.benchmark_v16.nml.IN
input.mom6_ccpp.nml.IN
input.mom6.nml.IN
input_nest02.nml.IN
multi_gases.nml.IN
regional_c768.nml.IN
regional.nml.IN
regional_rrtmgp.nml.IN
stretched-input.nml.IN
stretched-nest-input.nml.IN

junwang-noaa commented:
Moorthi, your branch has different CDEPS and WW3 revisions:

 8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d CDEPS-interface/CDEPS (cdeps0.4.4-3-g8d9b9ec)
 3740ba99caaa2822b5098bf32faeaf199a710a51 WW3 (6.07.1-97-g3740ba9)

while ufs-weather-model develop branch has:

 07ca4a33eb47be20629fdf30e72f2eaf690a5110 CDEPS-interface/CDEPS (cdeps0.4.2-24-g07ca4a3)
 097bbb65c4e372d52a733af911835a310cf54122 WW3 (6.07.1-88-g097bbb65)

Please point those two components to develop branch revisions. Thanks

SMoorthi-emc commented:
Is cdep develop or EMC/ develop?
I updated both cdep and ww3 last night.
Moorthi

Sent from my iPhone

> On Jun 22, 2021, at 4:31 PM, Jun Wang ***@***.***> wrote:
> 
> Moorthi, your branch has different CDEPS and WW3 revisions:
> 
> 8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d CDEPS-interface/CDEPS (cdeps0.4.4-3-g8d9b9ec)
> 3740ba99caaa2822b5098bf32faeaf199a710a51 WW3 (6.07.1-97-g3740ba9)
> 
> while ufs-weather-model develop branch has:
> 
> 07ca4a33eb47be20629fdf30e72f2eaf690a5110 CDEPS-interface/CDEPS (cdeps0.4.2-24-g07ca4a3)
> 097bbb65c4e372d52a733af911835a310cf54122 WW3 (6.07.1-88-g097bbb65)
> 
> Please point those two components to develop branch revisions. Thanks
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

junwang-noaa commented:
CDEPS is using the develop branch, the branch has a recent update, Bin Li
has a ufs-weather-model PR to update UFS to use that version. Without the
changes from that PR (or PR#611), the cdeps datm test can't get compiled.

On Tue, Jun 22, 2021 at 4:37 PM SMoorthi-emc ***@***.***>
wrote:

> Is cdep develop or EMC/ develop?
> I updated both cdep and ww3 last night.
> Moorthi
>
> Sent from my iPhone
>
> > On Jun 22, 2021, at 4:31 PM, Jun Wang ***@***.***> wrote:
> >
> > Moorthi, your branch has different CDEPS and WW3 revisions:
> >
> > 8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d CDEPS-interface/CDEPS
> (cdeps0.4.4-3-g8d9b9ec)
> > 3740ba99caaa2822b5098bf32faeaf199a710a51 WW3 (6.07.1-97-g3740ba9)
> >
> > while ufs-weather-model develop branch has:
> >
> > 07ca4a33eb47be20629fdf30e72f2eaf690a5110 CDEPS-interface/CDEPS
> (cdeps0.4.2-24-g07ca4a3)
> > 097bbb65c4e372d52a733af911835a310cf54122 WW3 (6.07.1-88-g097bbb65)
> >
> > Please point those two components to develop branch revisions. Thanks
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub, or unsubscribe.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866316517>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TPKGDIBTADUTP34UDDTUDYBPANCNFSM45U36N4Q>
> .
>

SMoorthi-emc commented:
CDEp/develop is at
"8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d"
Moorthi

On Tue, Jun 22, 2021 at 5:08 PM Jun Wang ***@***.***> wrote:

> CDEPS is using the develop branch, the branch has a recent update, Bin Li
> has a ufs-weather-model PR to update UFS to use that version. Without the
> changes from that PR (or PR#611), the cdeps datm test can't get compiled.
>
> On Tue, Jun 22, 2021 at 4:37 PM SMoorthi-emc ***@***.***>
> wrote:
>
> > Is cdep develop or EMC/ develop?
> > I updated both cdep and ww3 last night.
> > Moorthi
> >
> > Sent from my iPhone
> >
> > > On Jun 22, 2021, at 4:31 PM, Jun Wang ***@***.***> wrote:
> > >
> > > Moorthi, your branch has different CDEPS and WW3 revisions:
> > >
> > > 8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d CDEPS-interface/CDEPS
> > (cdeps0.4.4-3-g8d9b9ec)
> > > 3740ba99caaa2822b5098bf32faeaf199a710a51 WW3 (6.07.1-97-g3740ba9)
> > >
> > > while ufs-weather-model develop branch has:
> > >
> > > 07ca4a33eb47be20629fdf30e72f2eaf690a5110 CDEPS-interface/CDEPS
> > (cdeps0.4.2-24-g07ca4a3)
> > > 097bbb65c4e372d52a733af911835a310cf54122 WW3 (6.07.1-88-g097bbb65)
> > >
> > > Please point those two components to develop branch revisions. Thanks
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub, or unsubscribe.
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866316517
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AI7D6TPKGDIBTADUTP34UDDTUDYBPANCNFSM45U36N4Q
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866334861>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYXHJVTZUMV4GKBAGD3TUD3VDANCNFSM45U36N4Q>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

junwang-noaa commented:
Yes, that is the latest CDEPS develop revision, you need additional cmake
updates in ufs in order to use that version. Currently the CDEPS DATM tests
in your branch failed at compilation because of that.  Please point to the
CDEPS/develop revision  07ca4a3
<https://github.com/NOAA-EMC/CDEPS/tree/07ca4a33eb47be20629fdf30e72f2eaf690a5110>
that ufs develop branch currently uses

On Tue, Jun 22, 2021 at 7:34 PM SMoorthi-emc ***@***.***>
wrote:

> CDEp/develop is at
> "8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d"
> Moorthi
>
> On Tue, Jun 22, 2021 at 5:08 PM Jun Wang ***@***.***> wrote:
>
> > CDEPS is using the develop branch, the branch has a recent update, Bin Li
> > has a ufs-weather-model PR to update UFS to use that version. Without the
> > changes from that PR (or PR#611), the cdeps datm test can't get compiled.
> >
> > On Tue, Jun 22, 2021 at 4:37 PM SMoorthi-emc ***@***.***>
> > wrote:
> >
> > > Is cdep develop or EMC/ develop?
> > > I updated both cdep and ww3 last night.
> > > Moorthi
> > >
> > > Sent from my iPhone
> > >
> > > > On Jun 22, 2021, at 4:31 PM, Jun Wang ***@***.***> wrote:
> > > >
> > > > Moorthi, your branch has different CDEPS and WW3 revisions:
> > > >
> > > > 8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d CDEPS-interface/CDEPS
> > > (cdeps0.4.4-3-g8d9b9ec)
> > > > 3740ba99caaa2822b5098bf32faeaf199a710a51 WW3 (6.07.1-97-g3740ba9)
> > > >
> > > > while ufs-weather-model develop branch has:
> > > >
> > > > 07ca4a33eb47be20629fdf30e72f2eaf690a5110 CDEPS-interface/CDEPS
> > > (cdeps0.4.2-24-g07ca4a3)
> > > > 097bbb65c4e372d52a733af911835a310cf54122 WW3 (6.07.1-88-g097bbb65)
> > > >
> > > > Please point those two components to develop branch revisions. Thanks
> > > >
> > > > —
> > > > You are receiving this because you authored the thread.
> > > > Reply to this email directly, view it on GitHub, or unsubscribe.
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866316517
> > >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AI7D6TPKGDIBTADUTP34UDDTUDYBPANCNFSM45U36N4Q
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866334861
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ALLVRYXHJVTZUMV4GKBAGD3TUD3VDANCNFSM45U36N4Q
> >
> > .
> >
>
>
> --
> Dr. Shrinivas Moorthi
> Research Meteorologist
> Modeling and Data Assimilation Branch
> Environmental Modeling Center / National Centers for Environmental
> Prediction
> 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> Tel: (301)683-3718
>
> e-mail: ***@***.***
> Phone: (301) 683-3718 Fax: (301) 683-3718
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866405269>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TPNSP5TNWJL5YNU4ELTUEMXXANCNFSM45U36N4Q>
> .
>

SMoorthi-emc commented:
Well, I can't get the stochastic test working; it appears that allocated
arrays are reallocated again.
Moorthi

On Tue, Jun 22, 2021 at 9:46 PM Jun Wang ***@***.***> wrote:

> Yes, that is the latest CDEPS develop revision, you need additional cmake
> updates in ufs in order to use that version. Currently the CDEPS DATM tests
> in your branch failed at compilation because of that. Please point to the
> CDEPS/develop revision 07ca4a3
> <
> https://github.com/NOAA-EMC/CDEPS/tree/07ca4a33eb47be20629fdf30e72f2eaf690a5110
> >
> that ufs develop branch currently uses
>
> On Tue, Jun 22, 2021 at 7:34 PM SMoorthi-emc ***@***.***>
> wrote:
>
> > CDEp/develop is at
> > "8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d"
> > Moorthi
> >
> > On Tue, Jun 22, 2021 at 5:08 PM Jun Wang ***@***.***> wrote:
> >
> > > CDEPS is using the develop branch, the branch has a recent update, Bin
> Li
> > > has a ufs-weather-model PR to update UFS to use that version. Without
> the
> > > changes from that PR (or PR#611), the cdeps datm test can't get
> compiled.
> > >
> > > On Tue, Jun 22, 2021 at 4:37 PM SMoorthi-emc ***@***.***>
> > > wrote:
> > >
> > > > Is cdep develop or EMC/ develop?
> > > > I updated both cdep and ww3 last night.
> > > > Moorthi
> > > >
> > > > Sent from my iPhone
> > > >
> > > > > On Jun 22, 2021, at 4:31 PM, Jun Wang ***@***.***> wrote:
> > > > >
> > > > > Moorthi, your branch has different CDEPS and WW3 revisions:
> > > > >
> > > > > 8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d CDEPS-interface/CDEPS
> > > > (cdeps0.4.4-3-g8d9b9ec)
> > > > > 3740ba99caaa2822b5098bf32faeaf199a710a51 WW3 (6.07.1-97-g3740ba9)
> > > > >
> > > > > while ufs-weather-model develop branch has:
> > > > >
> > > > > 07ca4a33eb47be20629fdf30e72f2eaf690a5110 CDEPS-interface/CDEPS
> > > > (cdeps0.4.2-24-g07ca4a3)
> > > > > 097bbb65c4e372d52a733af911835a310cf54122 WW3 (6.07.1-88-g097bbb65)
> > > > >
> > > > > Please point those two components to develop branch revisions.
> Thanks
> > > > >
> > > > > —
> > > > > You are receiving this because you authored the thread.
> > > > > Reply to this email directly, view it on GitHub, or unsubscribe.
> > > >
> > > > —
> > > > You are receiving this because you commented.
> > > > Reply to this email directly, view it on GitHub
> > > > <
> > >
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866316517
> > > >,
> > > > or unsubscribe
> > > > <
> > >
> >
> https://github.com/notifications/unsubscribe-auth/AI7D6TPKGDIBTADUTP34UDDTUDYBPANCNFSM45U36N4Q
> > > >
> > > > .
> > > >
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866334861
> > >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ALLVRYXHJVTZUMV4GKBAGD3TUD3VDANCNFSM45U36N4Q
> > >
> > > .
> > >
> >
> >
> > --
> > Dr. Shrinivas Moorthi
> > Research Meteorologist
> > Modeling and Data Assimilation Branch
> > Environmental Modeling Center / National Centers for Environmental
> > Prediction
> > 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> > Tel: (301)683-3718
> >
> > e-mail: ***@***.***
> > Phone: (301) 683-3718 Fax: (301) 683-3718
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866405269
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AI7D6TPNSP5TNWJL5YNU4ELTUEMXXANCNFSM45U36N4Q
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866457266>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYUO2Y7RMWQRZUDVP3DTUE4HRANCNFSM45U36N4Q>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

junwang-noaa commented:
So this is the control_c384gdas case?

On Tue, Jun 22, 2021 at 10:04 PM SMoorthi-emc ***@***.***>
wrote:

> Well, I can't get the stochastic test working; it appears that allocated
> arrays are reallocated again.
> Moorthi
>
> On Tue, Jun 22, 2021 at 9:46 PM Jun Wang ***@***.***> wrote:
>
> > Yes, that is the latest CDEPS develop revision, you need additional cmake
> > updates in ufs in order to use that version. Currently the CDEPS DATM
> tests
> > in your branch failed at compilation because of that. Please point to the
> > CDEPS/develop revision 07ca4a3
> > <
> >
> https://github.com/NOAA-EMC/CDEPS/tree/07ca4a33eb47be20629fdf30e72f2eaf690a5110
> > >
> > that ufs develop branch currently uses
> >
> > On Tue, Jun 22, 2021 at 7:34 PM SMoorthi-emc ***@***.***>
> > wrote:
> >
> > > CDEp/develop is at
> > > "8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d"
> > > Moorthi
> > >
> > > On Tue, Jun 22, 2021 at 5:08 PM Jun Wang ***@***.***> wrote:
> > >
> > > > CDEPS is using the develop branch, the branch has a recent update,
> Bin
> > Li
> > > > has a ufs-weather-model PR to update UFS to use that version. Without
> > the
> > > > changes from that PR (or PR#611), the cdeps datm test can't get
> > compiled.
> > > >
> > > > On Tue, Jun 22, 2021 at 4:37 PM SMoorthi-emc ***@***.***>
> > > > wrote:
> > > >
> > > > > Is cdep develop or EMC/ develop?
> > > > > I updated both cdep and ww3 last night.
> > > > > Moorthi
> > > > >
> > > > > Sent from my iPhone
> > > > >
> > > > > > On Jun 22, 2021, at 4:31 PM, Jun Wang ***@***.***> wrote:
> > > > > >
> > > > > > Moorthi, your branch has different CDEPS and WW3 revisions:
> > > > > >
> > > > > > 8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d CDEPS-interface/CDEPS
> > > > > (cdeps0.4.4-3-g8d9b9ec)
> > > > > > 3740ba99caaa2822b5098bf32faeaf199a710a51 WW3 (6.07.1-97-g3740ba9)
> > > > > >
> > > > > > while ufs-weather-model develop branch has:
> > > > > >
> > > > > > 07ca4a33eb47be20629fdf30e72f2eaf690a5110 CDEPS-interface/CDEPS
> > > > > (cdeps0.4.2-24-g07ca4a3)
> > > > > > 097bbb65c4e372d52a733af911835a310cf54122 WW3
> (6.07.1-88-g097bbb65)
> > > > > >
> > > > > > Please point those two components to develop branch revisions.
> > Thanks
> > > > > >
> > > > > > —
> > > > > > You are receiving this because you authored the thread.
> > > > > > Reply to this email directly, view it on GitHub, or unsubscribe.
> > > > >
> > > > > —
> > > > > You are receiving this because you commented.
> > > > > Reply to this email directly, view it on GitHub
> > > > > <
> > > >
> > >
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866316517
> > > > >,
> > > > > or unsubscribe
> > > > > <
> > > >
> > >
> >
> https://github.com/notifications/unsubscribe-auth/AI7D6TPKGDIBTADUTP34UDDTUDYBPANCNFSM45U36N4Q
> > > > >
> > > > > .
> > > > >
> > > >
> > > > —
> > > > You are receiving this because you authored the thread.
> > > > Reply to this email directly, view it on GitHub
> > > > <
> > >
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866334861
> > > >,
> > > > or unsubscribe
> > > > <
> > >
> >
> https://github.com/notifications/unsubscribe-auth/ALLVRYXHJVTZUMV4GKBAGD3TUD3VDANCNFSM45U36N4Q
> > > >
> > > > .
> > > >
> > >
> > >
> > > --
> > > Dr. Shrinivas Moorthi
> > > Research Meteorologist
> > > Modeling and Data Assimilation Branch
> > > Environmental Modeling Center / National Centers for Environmental
> > > Prediction
> > > 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> > > Tel: (301)683-3718
> > >
> > > e-mail: ***@***.***
> > > Phone: (301) 683-3718 Fax: (301) 683-3718
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866405269
> > >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AI7D6TPNSP5TNWJL5YNU4ELTUEMXXANCNFSM45U36N4Q
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866457266
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ALLVRYUO2Y7RMWQRZUDVP3DTUE4HRANCNFSM45U36N4Q
> >
> > .
> >
>
>
> --
> Dr. Shrinivas Moorthi
> Research Meteorologist
> Modeling and Data Assimilation Branch
> Environmental Modeling Center / National Centers for Environmental
> Prediction
> 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> Tel: (301)683-3718
>
> e-mail: ***@***.***
> Phone: (301) 683-3718 Fax: (301) 683-3718
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866463876>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TOCIDUIOZIM52DMK53TUE6LTANCNFSM45U36N4Q>
> .
>

SMoorthi-emc commented:
I am not sure why this is happening.  I know there are two calls to
"stochastic_wrapper" in atmos_model.F90.
I am not sure how it is working in the regression tests.  I fixed some
allocations by deallocating, but it is a rabbit hole.
Allocated arrays keep appearing.  I shouldn't have to change stochastic
physics.
I am giving up for now.
Moorthi

On Tue, Jun 22, 2021 at 10:04 PM Shrinivas Moorthi - NOAA Federal <
***@***.***> wrote:

> Well, I can't get the stochastic test working; it appears that allocated
> arrays are reallocated again.
> Moorthi
>
> On Tue, Jun 22, 2021 at 9:46 PM Jun Wang ***@***.***> wrote:
>
>> Yes, that is the latest CDEPS develop revision, you need additional cmake
>> updates in ufs in order to use that version. Currently the CDEPS DATM
>> tests
>> in your branch failed at compilation because of that. Please point to the
>> CDEPS/develop revision 07ca4a3
>> <
>> https://github.com/NOAA-EMC/CDEPS/tree/07ca4a33eb47be20629fdf30e72f2eaf690a5110
>> >
>> that ufs develop branch currently uses
>>
>> On Tue, Jun 22, 2021 at 7:34 PM SMoorthi-emc ***@***.***>
>> wrote:
>>
>> > CDEp/develop is at
>> > "8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d"
>> > Moorthi
>> >
>> > On Tue, Jun 22, 2021 at 5:08 PM Jun Wang ***@***.***> wrote:
>> >
>> > > CDEPS is using the develop branch, the branch has a recent update,
>> Bin Li
>> > > has a ufs-weather-model PR to update UFS to use that version. Without
>> the
>> > > changes from that PR (or PR#611), the cdeps datm test can't get
>> compiled.
>> > >
>> > > On Tue, Jun 22, 2021 at 4:37 PM SMoorthi-emc ***@***.***>
>> > > wrote:
>> > >
>> > > > Is cdep develop or EMC/ develop?
>> > > > I updated both cdep and ww3 last night.
>> > > > Moorthi
>> > > >
>> > > > Sent from my iPhone
>> > > >
>> > > > > On Jun 22, 2021, at 4:31 PM, Jun Wang ***@***.***> wrote:
>> > > > >
>> > > > > Moorthi, your branch has different CDEPS and WW3 revisions:
>> > > > >
>> > > > > 8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d CDEPS-interface/CDEPS
>> > > > (cdeps0.4.4-3-g8d9b9ec)
>> > > > > 3740ba99caaa2822b5098bf32faeaf199a710a51 WW3 (6.07.1-97-g3740ba9)
>> > > > >
>> > > > > while ufs-weather-model develop branch has:
>> > > > >
>> > > > > 07ca4a33eb47be20629fdf30e72f2eaf690a5110 CDEPS-interface/CDEPS
>> > > > (cdeps0.4.2-24-g07ca4a3)
>> > > > > 097bbb65c4e372d52a733af911835a310cf54122 WW3 (6.07.1-88-g097bbb65)
>> > > > >
>> > > > > Please point those two components to develop branch revisions.
>> Thanks
>> > > > >
>> > > > > —
>> > > > > You are receiving this because you authored the thread.
>> > > > > Reply to this email directly, view it on GitHub, or unsubscribe.
>> > > >
>> > > > —
>> > > > You are receiving this because you commented.
>> > > > Reply to this email directly, view it on GitHub
>> > > > <
>> > >
>> >
>> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866316517
>> > > >,
>> > > > or unsubscribe
>> > > > <
>> > >
>> >
>> https://github.com/notifications/unsubscribe-auth/AI7D6TPKGDIBTADUTP34UDDTUDYBPANCNFSM45U36N4Q
>> > > >
>> > > > .
>> > > >
>> > >
>> > > —
>> > > You are receiving this because you authored the thread.
>> > > Reply to this email directly, view it on GitHub
>> > > <
>> >
>> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866334861
>> > >,
>> > > or unsubscribe
>> > > <
>> >
>> https://github.com/notifications/unsubscribe-auth/ALLVRYXHJVTZUMV4GKBAGD3TUD3VDANCNFSM45U36N4Q
>> > >
>> > > .
>> > >
>> >
>> >
>> > --
>> > Dr. Shrinivas Moorthi
>> > Research Meteorologist
>> > Modeling and Data Assimilation Branch
>> > Environmental Modeling Center / National Centers for Environmental
>> > Prediction
>> > 5830 University Research Court - (W/NP23), College Park MD 20740 USA
>> > Tel: (301)683-3718
>> >
>> > e-mail: ***@***.***
>> > Phone: (301) 683-3718 Fax: (301) 683-3718
>> >
>> > —
>> > You are receiving this because you commented.
>> > Reply to this email directly, view it on GitHub
>> > <
>> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866405269
>> >,
>> > or unsubscribe
>> > <
>> https://github.com/notifications/unsubscribe-auth/AI7D6TPNSP5TNWJL5YNU4ELTUEMXXANCNFSM45U36N4Q
>> >
>> > .
>> >
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866457266>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ALLVRYUO2Y7RMWQRZUDVP3DTUE4HRANCNFSM45U36N4Q>
>> .
>>
>
>
> --
> Dr. Shrinivas Moorthi
> Research Meteorologist
> Modeling and Data Assimilation Branch
> Environmental Modeling Center / National Centers for Environmental
> Prediction
> 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> Tel: (301)683-3718
>
> e-mail: ***@***.***
> Phone:  (301) 683-3718   Fax: (301) 683-3718
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

SMoorthi-emc commented:
Yes.  I had set "kdt" to the actual "kdt" rather than zero (which is in the
current develop).
However, in the stochastic wrapper, I defined a local kdt by subtracting
the starting kdt (in other word, in the wrapper it would look like old kdt),
but that did not solve the issue. I don't know why the init routines are
getting called twice (I am guessing that is why it is complaining about
allocations).

On Tue, Jun 22, 2021 at 10:09 PM Jun Wang ***@***.***> wrote:

> So this is the control_c384gdas case?
>
> On Tue, Jun 22, 2021 at 10:04 PM SMoorthi-emc ***@***.***>
> wrote:
>
> > Well, I can't get the stochastic test working; it appears that allocated
> > arrays are reallocated again.
> > Moorthi
> >
> > On Tue, Jun 22, 2021 at 9:46 PM Jun Wang ***@***.***> wrote:
> >
> > > Yes, that is the latest CDEPS develop revision, you need additional
> cmake
> > > updates in ufs in order to use that version. Currently the CDEPS DATM
> > tests
> > > in your branch failed at compilation because of that. Please point to
> the
> > > CDEPS/develop revision 07ca4a3
> > > <
> > >
> >
> https://github.com/NOAA-EMC/CDEPS/tree/07ca4a33eb47be20629fdf30e72f2eaf690a5110
> > > >
> > > that ufs develop branch currently uses
> > >
> > > On Tue, Jun 22, 2021 at 7:34 PM SMoorthi-emc ***@***.***>
> > > wrote:
> > >
> > > > CDEp/develop is at
> > > > "8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d"
> > > > Moorthi
> > > >
> > > > On Tue, Jun 22, 2021 at 5:08 PM Jun Wang ***@***.***> wrote:
> > > >
> > > > > CDEPS is using the develop branch, the branch has a recent update,
> > Bin
> > > Li
> > > > > has a ufs-weather-model PR to update UFS to use that version.
> Without
> > > the
> > > > > changes from that PR (or PR#611), the cdeps datm test can't get
> > > compiled.
> > > > >
> > > > > On Tue, Jun 22, 2021 at 4:37 PM SMoorthi-emc ***@***.***>
> > > > > wrote:
> > > > >
> > > > > > Is cdep develop or EMC/ develop?
> > > > > > I updated both cdep and ww3 last night.
> > > > > > Moorthi
> > > > > >
> > > > > > Sent from my iPhone
> > > > > >
> > > > > > > On Jun 22, 2021, at 4:31 PM, Jun Wang ***@***.***> wrote:
> > > > > > >
> > > > > > > Moorthi, your branch has different CDEPS and WW3 revisions:
> > > > > > >
> > > > > > > 8d9b9ecfdcb790bc19a58a9611a27bd43c9ebd2d CDEPS-interface/CDEPS
> > > > > > (cdeps0.4.4-3-g8d9b9ec)
> > > > > > > 3740ba99caaa2822b5098bf32faeaf199a710a51 WW3
> (6.07.1-97-g3740ba9)
> > > > > > >
> > > > > > > while ufs-weather-model develop branch has:
> > > > > > >
> > > > > > > 07ca4a33eb47be20629fdf30e72f2eaf690a5110 CDEPS-interface/CDEPS
> > > > > > (cdeps0.4.2-24-g07ca4a3)
> > > > > > > 097bbb65c4e372d52a733af911835a310cf54122 WW3
> > (6.07.1-88-g097bbb65)
> > > > > > >
> > > > > > > Please point those two components to develop branch revisions.
> > > Thanks
> > > > > > >
> > > > > > > —
> > > > > > > You are receiving this because you authored the thread.
> > > > > > > Reply to this email directly, view it on GitHub, or
> unsubscribe.
> > > > > >
> > > > > > —
> > > > > > You are receiving this because you commented.
> > > > > > Reply to this email directly, view it on GitHub
> > > > > > <
> > > > >
> > > >
> > >
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866316517
> > > > > >,
> > > > > > or unsubscribe
> > > > > > <
> > > > >
> > > >
> > >
> >
> https://github.com/notifications/unsubscribe-auth/AI7D6TPKGDIBTADUTP34UDDTUDYBPANCNFSM45U36N4Q
> > > > > >
> > > > > > .
> > > > > >
> > > > >
> > > > > —
> > > > > You are receiving this because you authored the thread.
> > > > > Reply to this email directly, view it on GitHub
> > > > > <
> > > >
> > >
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866334861
> > > > >,
> > > > > or unsubscribe
> > > > > <
> > > >
> > >
> >
> https://github.com/notifications/unsubscribe-auth/ALLVRYXHJVTZUMV4GKBAGD3TUD3VDANCNFSM45U36N4Q
> > > > >
> > > > > .
> > > > >
> > > >
> > > >
> > > > --
> > > > Dr. Shrinivas Moorthi
> > > > Research Meteorologist
> > > > Modeling and Data Assimilation Branch
> > > > Environmental Modeling Center / National Centers for Environmental
> > > > Prediction
> > > > 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> > > > Tel: (301)683-3718
> > > >
> > > > e-mail: ***@***.***
> > > > Phone: (301) 683-3718 Fax: (301) 683-3718
> > > >
> > > > —
> > > > You are receiving this because you commented.
> > > > Reply to this email directly, view it on GitHub
> > > > <
> > >
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866405269
> > > >,
> > > > or unsubscribe
> > > > <
> > >
> >
> https://github.com/notifications/unsubscribe-auth/AI7D6TPNSP5TNWJL5YNU4ELTUEMXXANCNFSM45U36N4Q
> > > >
> > > > .
> > > >
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866457266
> > >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ALLVRYUO2Y7RMWQRZUDVP3DTUE4HRANCNFSM45U36N4Q
> > >
> > > .
> > >
> >
> >
> > --
> > Dr. Shrinivas Moorthi
> > Research Meteorologist
> > Modeling and Data Assimilation Branch
> > Environmental Modeling Center / National Centers for Environmental
> > Prediction
> > 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> > Tel: (301)683-3718
> >
> > e-mail: ***@***.***
> > Phone: (301) 683-3718 Fax: (301) 683-3718
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866463876
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AI7D6TOCIDUIOZIM52DMK53TUE6LTANCNFSM45U36N4Q
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-866465850>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYQ6MJ3S6CZAGHKSVRTTUE64RANCNFSM45U36N4Q>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

DeniseWorthen commented on tests/tests/fv3_rap:
```diff
@@ -73,6 +73,7 @@ export LIST_FILES="atmos_4xdaily.tile1.nc \
 
 export_fv3
 
+export RAP=true
```
Does this need to be a logical (.true.)?

arunchawla-NOAA commented:
This PR does not contain a description of what it solves and no connection to what issues are being resolved.  This provides no record of what was fixed. Please update before the PR can be worked on

SMoorthi-emc commented:
Thanks for catching this.  It is not needed (it was used to change
diag_table while debugging), so I have removed it.

On Thu, Jul 8, 2021 at 9:00 AM Denise Worthen ***@***.***>
wrote:

> *@DeniseWorthen* commented on this pull request.
> ------------------------------
>
> In tests/tests/fv3_rap
> <https://github.com/ufs-community/ufs-weather-model/pull/606#discussion_r666169661>
> :
>
> > @@ -73,6 +73,7 @@ export LIST_FILES="atmos_4xdaily.tile1.nc \
>
>  export_fv3
>
> +export RAP=true
>
> Does this need to be a logical (.true.)?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#pullrequestreview-702030639>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYX4ZGU5RAJZ35PKISLTWWONVANCNFSM45U36N4Q>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

SMoorthi-emc commented:
Arun Chawla, the description is provided in
"
    CCPP-Physics Pull request Sm may182021 #662
    fv3atm full request may182021 #303
"
It is unfair to expect details in every PR - a consequence of having multiple repositories for a single model

arunchawla-NOAA commented:
We need to be able to have a document trail. You can copy and paste the description from the other PRs. You should also list the issues that are being solved by the PR and the testing. The template has been provided to minimize the effort needed.

SMoorthi-emc commented:
At the ufs-weather-model level this PR fixes errors in two regression tests - "control_ras" and "control_ras_debug".
Additionally this PR completes implementation of upgrades in  PRs
"CCPP-Physics Pull request Sm may182021 #662
fv3atm full request may182021 #303"
These two PRs include improvements to "sfcsub.F" and "gcycle.F90" in ccpp-physics to handle the fractional grid correctly (i.e. providing seasonal variability of land variables for the land portion while providing the same for surface temperature over open water).  It also optimizes the interpolation of Merra2 aerosols and the driver of flake model.  Additionally, the surface compositing is improved by treating snow depth over ice and land and land snow depth is saved in the surface restart file (updates in FV3GFS_io.F90 which also includes additional cleanup and a bug fix that makes it thread safe)
Stochastic physics wrapper is updated to make it independent of "kdt" and cleaned up.
Stochastic physics wrapper not called unless some aspects of stochastic physics or Cellular Automata are turned on.
The variable "ncld" is removed from the ATM namelist as this information is redundant to "imp_physics".
This required changes to most regression tests.
The emissivity for ice and water in surface_composite is made consistent with that in radiation_surface.

SMoorthi-emc commented:
This PR also should fix an outstanding issue in the current parallel benchmark runs by adding lake-ice seasonal cycle via climatology.

SMoorthi-emc commented:
Removed from compile.sh


On Thu, Jul 8, 2021 at 1:47 PM Brian Curtis ***@***.***>
wrote:

> *@BrianCurtis-NOAA* commented on this pull request.
> ------------------------------
>
> In tests/compile.sh
> <https://github.com/ufs-community/ufs-weather-model/pull/606#discussion_r666400969>
> :
>
> > @@ -54,6 +54,7 @@ BUILD_JOBS=${BUILD_JOBS:-8}
>
>  hostname
>
> +module purge
>
> This is still an issue @SMoorthi-emc <https://github.com/SMoorthi-emc> we
> can't module purge on all machines, it breaks Gaea for example.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#discussion_r666400969>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYRQDAIJXBOBJ2POAJTTWXQDHANCNFSM45U36N4Q>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

SMoorthi-emc commented:
Hello,
   After I removed "module purge" from compile.sh, I  can't compile on WCOSS/Dell anymore.
Can someone help how to overcome this or I should just stick with having the "module purge" in my version of compile.sh?
Thanks

SMoorthi-emc commented:
I am not sure how this would work.  I can't see the path.  Which script
calls this script?
compile.sh calls the modulefile "ufs_$MACHINE_ID" in the very beginning.

On Wed, Jun 2, 2021 at 5:01 PM Jun Wang ***@***.***> wrote:

> I think it's this script for module purge.
>
> source $PATHTR/NEMS/src/conf/module-setup.sh.inc
>
> On Wed, Jun 2, 2021 at 3:54 PM Dom Heinzeller ***@***.***>
> wrote:
>
> > ***@***.**** commented on this pull request.
> > ------------------------------
> >
> > In tests/compile.sh
> > <
> https://github.com/ufs-community/ufs-weather-model/pull/606#discussion_r644276646
> >
> > :
> >
> > > @@ -54,6 +54,7 @@ BUILD_JOBS=${BUILD_JOBS:-8}
> >
> > hostname
> >
> > +module purge
> >
> > No, this must be in machine specific files. If you do module purge on
> > gaea, for example, the whole system falls apart.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/pull/606#pullrequestreview-674624688
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AI7D6TPYAY5QFJXF34HEMITTQ2EAJANCNFSM45U36N4Q
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-853379475>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYUPIS7CCWGR2IRCARTTQ2LYTANCNFSM45U36N4Q>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

climbfuji commented on tests/compile.sh:
```diff
@@ -48,6 +48,8 @@ if [[ $MACHINE_ID == cheyenne.* ]] ; then
     BUILD_JOBS=${BUILD_JOBS:-3}
 elif [[ $MACHINE_ID == wcoss_dell_p3 ]] ; then
     BUILD_JOBS=${BUILD_JOBS:-1}
+    . $MODULESHOME/init/sh
```
Really not sure why this is needed, since the logic in `NEMS/src/conf/module-setup.*` is:
```
else if ( { test -L /usrx && sh -c "readlink /usrx 2> /dev/null | grep dell" } ) then
    # We are on WCOSS Mars or Venus
    if ( ! { module help >& /dev/null } ) then
        source /usrx/local/prod/lmod/lmod/init/$__ms_shell
    endif
    module purge
```

junwang-noaa replied:
I would suggest changing these two lines to: 
source ${PATHTR}/NEMS/src/conf/module-setup.sh.inc

SMoorthi-emc commented:
I can’t compile on dell without purge.
Moorthi

Sent from my iPhone

> On Jul 12, 2021, at 3:55 PM, Dom Heinzeller ***@***.***> wrote:
> 
> @climbfuji commented on this pull request.
> 
> In tests/compile.sh:
> 
> > @@ -48,6 +48,8 @@ if [[ $MACHINE_ID == cheyenne.* ]] ; then
>      BUILD_JOBS=${BUILD_JOBS:-3}
>  elif [[ $MACHINE_ID == wcoss_dell_p3 ]] ; then
>      BUILD_JOBS=${BUILD_JOBS:-1}
> +    . $MODULESHOME/init/sh
> Really not sure why this is needed, since the logic in NEMS/src/conf/module-setup.* is:
> 
> else if ( { test -L /usrx && sh -c "readlink /usrx 2> /dev/null | grep dell" } ) then
>     # We are on WCOSS Mars or Venus
>     if ( ! { module help >& /dev/null } ) then
>         source /usrx/local/prod/lmod/lmod/init/$__ms_shell
>     endif
>     module purge
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

climbfuji commented:
> I can’t compile on dell without purge. Moorthi
> […](#)

Let me try to reproduce the problem.

SMoorthi-emc commented:
It could be my problem as my .bashrc loads modules.
Moorrhi

Sent from my iPhone

> On Jul 12, 2021, at 4:26 PM, Dom Heinzeller ***@***.***> wrote:
> 
> I can’t compile on dell without purge. Moorthi
> …
> 
> Let me try to reproduce the problem.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

DeniseWorthen commented:
The code in atmos_model.F90 places artificial limits on imported ice thickness or snow thickness. 

I've approved the PR, but I do not think this is a good idea at this stage of development and testing of the fully coupled model. If these values are unreasonable, that is something the evaluators of the coupled model need to be able to see. Or, if it causes the model to fail, then we need to know that. We don't want to be hiding bugs or potential issues w/ the mapping. Now is the time to spot and correct the issue if possible.

SMoorthi-emc commented:
Since some of the regression tests have been done, it is not good to relax
this now.
I will have another PR soon (it is already working) to update CIC6 albedo
back to FV3.  At that time I could further relax those parameters.
Moorthi

On Mon, Jul 12, 2021 at 5:21 PM Denise Worthen ***@***.***>
wrote:

> I've approved the PR, but I do not think we should be placing artificial
> limits on imported ice thickness or snow thickness at this stage of
> development and testing of the fully coupled model. If these values are
> unreasonable, that is something the evaluators of the coupled model need to
> be able to see. Or, if it causes the model to fail, then we need to know
> that. We don't want to be hiding bugs or potential issues w/ the mapping.
> Now is the time to spot and correct the issue if possible.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/606#issuecomment-878605042>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYQRDU7EK7G7BZQXZLLTXNMGBANCNFSM45U36N4Q>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718
